app/assets/javascripts/discourse/app/components/modal/bulk-topic-actions.gjs

```diff
@@ -0,0 +1,153 @@
+
+  async perform(operation) {
+    this.loading = true;
+
+    if (this.args.model.topics.length > 20) {
+      this.showProgress = true;
+    }
+
+    try {
+      return this._processChunks(operation);
+    } catch {
+      this.dialog.alert(i18n.t("generic_error"));
+    } finally {
+      this.loading = false;
+      this.processedTopicCount = 0;
+      this.showProgress = false;
+    }
+  }
+
+  _generateTopicChunks(allTopics) {
+    let startIndex = 0;
+    const chunkSize = 30;
+    const chunks = [];
+
+    while (startIndex < allTopics.length) {
+      const topics = allTopics.slice(startIndex, startIndex + chunkSize);
+      chunks.push(topics);
+      startIndex += chunkSize;
+    }
+
+    return chunks;
+  }
+
+  _processChunks(operation) {
+    const allTopics = this.args.model.topics;
+    const topicChunks = this._generateTopicChunks(allTopics);
+    const topicIds = [];
+
+    const tasks = topicChunks.map((topics) => async () => {
+      const result = await Topic.bulkOperation(topics, operation);
+      this.processedTopicCount = this.processedTopicCount + topics.length;
+      return result;
+    });
+
+    return new Promise((resolve, reject) => {
+      const resolveNextTask = async () => {
+        if (tasks.length === 0) {
+          const topics = topicIds.map((id) => allTopics.findBy("id", id));
+          return resolve(topics);
+        }
+
+        const task = tasks.shift();
+
+        try {
+          const result = await task();
+          if (result?.topic_ids) {
+            topicIds.push(...result.topic_ids);
+          }
+          resolveNextTask();
+        } catch {
+          reject();
+        }
+      };
+
+      resolveNextTask();
+    });
+  }
+
+  @action
+  setComponent(component) {
+    this.activeComponent = component;
+  }
+
+  @action
+  performAction() {
+    // TODO: return only visible topics
+    //visible: ({ topics }) => !topics.some((t) => t.isPrivateMessage),
+
+    switch (this.args.model.action) {
+      case "close":
+        this.forEachPerformed({ type: "close" }, (t) => t.set("closed", true));
+        break;
+    }
+  }
+
+  @action
+  async forEachPerformed(operation, cb) {
+    const topics = await this.perform(operation);
+
+    if (topics) {
+      topics.forEach(cb);
+      this.args.model.refreshClosure?.();
+      this.args.closeModal();
+
+      window.location.reload();
+    }
+  }
+
+  @action
+  async performAndRefresh(operation) {
+    await this.perform(operation);
+
+    this.args.model.refreshClosure?.();
+    this.args.closeModal();
+  }
+
+  <template>
+    <DModal
+      @title={{@model.title}}
+      @closeModal={{@closeModal}}
+      class="topic-bulk-actions-modal -large"
+    >
+      <:body>
+        <div>
+          {{htmlSafe (i18n "topics.bulk.selected" count=@model.topics.length)}}
+        </div>
+        <div>body</div>
+      </:body>
+
+      <:footer>
+        {{#if @model.silent}}
+          <div><input class="" id="silent" type="checkbox" />
```

Comment:
Slight change in structure improves how this input looks:

```html
<div class="topic-bulk-action-options">
    <label for="topic-bulk-action-options__silent" class="checkbox-label">
    <input id="topic-bulk-action-options__silent" type="checkbox"> Perform this action silently.</label>
</div>
```

Then the `topic-bulk-action-options` CSS class needs `display: flex`, and the Confirm button needs `margin-left: auto`

![image](https://github.com/discourse/discourse/assets/920448/d5dd2e83-11b9-4518-87c4-48d0017fb106)

Reply:
Also we are not binding this `silent` checkbox to anything, it needs to be sent as an the `silent: true` option to `Topic.bulkOperation`  which is used further down the line in `TopicStatusUpdater`